.github/workflows/publish-on-tag.yml

```diff
@@ -0,0 +1,44 @@
+name: Publish the package when a tag is pushed
+
+on:
+  push:
+    tags:
+      - 'v*.*.*' # Trigger only for tags starting with 'v' (e.g., v0.1.9)
+    branches:
+      - 'docker*' # Trigger for pushes to 'docker*' branches for testing
+env:
+  IMAGE_NAME: ghcr.io/marcua/ayb
+
+jobs:
+  build-and-push:
+    name: Build and Push Docker Image
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ env.IMAGE_NAME }}
+        tags: |
+          type=match,pattern=(v\d+\.\d+\.\d+),group=1
+          type=raw,value=latest
+          type=sha,prefix=,format=short
+
+    - name: Log in to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build Docker image
+      run: |
+        docker build -t ${{ env.IMAGE_NAME }}:${{ github.ref_name }} .
```

Comment:
docker tag is with ref_name (branch or git tag), but doesn't take advantage of lines 27-30